src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1199,6 +1193,23 @@
     <Field Name="Label" Type="LabelSymbol" Null="disallow"/>
   </Node>
 
+  <Node Name="BoundUnconvertedSwitchExpression" Base="BoundSwitchExpression">
+  </Node>
+
+  <!-- 
+  Switch expressions can convert to the target type.
+  Once converted to a target type, they cannot be target-typed again. 
+  The Converted switch expression is one which has been already converted to a target type.
+  Converted switch expressions always has a type.
+  -->
+  <Node Name="BoundConvertedSwitchExpression" Base="BoundSwitchExpression">
+    <!-- Converted switch expression must have a type, even if that's `void` for a statement. -->
```

Comment:
>even if that's `void` for a statement. [](start = 55, length = 38)

Switch expressions can't be statements?

Reply:
I think switch expressions cannot be used in expression statements at the moment.

---
In reply to: [301350455](https://github.com/dotnet/roslyn/pull/37052#discussion_r301350455) [](ancestors = 301350455)

Reply:
Correct.

---
In reply to: [301716937](https://github.com/dotnet/roslyn/pull/37052#discussion_r301716937) [](ancestors = 301716937,301350455)